src/platform/graphics/uniform-buffer.js

```diff
@@ -1,69 +1,33 @@
 import { Debug } from '../../core/debug.js';
 import {
     uniformTypeToName,
-    UNIFORMTYPE_INT, UNIFORMTYPE_FLOAT, UNIFORMTYPE_VEC2, UNIFORMTYPE_VEC3,
+    UNIFORMTYPE_INT, UNIFORMTYPE_FLOAT, UNIFORMTYPE_BOOL, UNIFORMTYPE_VEC2,
+    UNIFORMTYPE_VEC3, UNIFORMTYPE_UINT, UNIFORMTYPE_UVEC3, UNIFORMTYPE_UVEC4,
     UNIFORMTYPE_VEC4, UNIFORMTYPE_IVEC2, UNIFORMTYPE_IVEC3, UNIFORMTYPE_IVEC4,
     UNIFORMTYPE_FLOATARRAY, UNIFORMTYPE_VEC2ARRAY, UNIFORMTYPE_VEC3ARRAY,
-    UNIFORMTYPE_MAT2, UNIFORMTYPE_MAT3
+    UNIFORMTYPE_MAT2, UNIFORMTYPE_MAT3, UNIFORMTYPE_UVEC2, UNIFORMTYPE_IVEC2ARRAY, UNIFORMTYPE_IVEC3ARRAY, UNIFORMTYPE_IVEC4ARRAY, UNIFORMTYPE_UVEC2ARRAY, UNIFORMTYPE_UVEC3ARRAY, UNIFORMTYPE_UVEC4ARRAY, UNIFORMTYPE_VEC4ARRAY, UNIFORMTYPE_BVEC2, UNIFORMTYPE_BVEC4, UNIFORMTYPE_BVEC3, UNIFORMTYPE_BVEC2ARRAY, UNIFORMTYPE_BVEC3ARRAY, UNIFORMTYPE_BVEC4ARRAY
 } from './constants.js';
 import { DynamicBufferAllocation } from './dynamic-buffers.js';
 
+function updateSingleElement(uniformBuffer, value, offset, numComponents) {
+    const dst = uniformBuffer.getStorageForType(value.type);
+    for (let i = 0; i < numComponents; i++) {
+        dst[offset + i] = value[i];
+    }
+}
+
+function updateArray(uniformBuffer, value, offset, numComponents, count) {
+    const dst = uniformBuffer.getStorageForType(value.type);
+    for (let i = 0; i < count; i++) {
+        for (let j = 0; j < numComponents; j++) {
+            dst[offset + i * numComponents + j] = value[i * numComponents + j];
```

Comment:
I have not checked the specs, but I thought there was an alignment (padding) to follow as well here? Similarly to how UNIFORMTYPE_MAT3 format is handled ... vec3 is padded to vec4 and similar rules.